test/suite/test_cc09.py

```diff
@@ -0,0 +1,109 @@
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+from eviction_util import eviction_util
+from test_cc01 import test_cc_base
+from wiredtiger import stat
+from wtscenario import make_scenarios
+
+# test_cc09.py
+# Verify checkpoint cleanup reads pages from the disk to remove any obsolete time window information
+# present on the page.
+class test_cc09(eviction_util, test_cc_base):
+    conn_config_common = 'statistics=(all),statistics_log=(json,wait=1,on_close=true),verbose=(checkpoint_cleanup:0)'
+
+    # These settings set a limit to the number of btrees/pages that can be cleaned up per btree per
+    # checkpoint by the checkpoint cleanup thread.
+    conn_config_values = [
+        ('no_btrees', dict(expected_cleanup=False, cc_obsolete_tw_max=0, conn_config=f'{conn_config_common},heuristic_controls=[obsolete_tw_btree_max=0]')),
+        ('no_pages', dict(expected_cleanup=False, cc_obsolete_tw_max=0, conn_config=f'{conn_config_common},heuristic_controls=[checkpoint_cleanup_obsolete_tw_pages_dirty_max=0]')),
+        ('50_pages', dict(expected_cleanup=True, cc_obsolete_tw_max=50, conn_config=f'{conn_config_common},heuristic_controls=[checkpoint_cleanup_obsolete_tw_pages_dirty_max=50]')),
+        ('100_pages', dict(expected_cleanup=True, cc_obsolete_tw_max=100, conn_config=f'{conn_config_common},heuristic_controls=[checkpoint_cleanup_obsolete_tw_pages_dirty_max=100]')),
+        ('500_pages', dict(expected_cleanup=True, cc_obsolete_tw_max=500, conn_config=f'{conn_config_common},heuristic_controls=[checkpoint_cleanup_obsolete_tw_pages_dirty_max=500]')),
+    ]
+
+    # Necessary conditions for checkpoint cleanup to run.
+    cc_scenarios = [
+        ('newest_stop_durable_ts', dict(has_delete=True, bump_oldest_ts=False)),
+        ('obsolete_ts', dict(has_delete=False, bump_oldest_ts=True)),
+        ('none', dict(has_delete=False, bump_oldest_ts=False)),
+    ]
+
+    scenarios = make_scenarios(conn_config_values, cc_scenarios)
+
+    def test_evict(self):
```

Comment:
The function name can be CC instead of evict.

Reply:
Nice finding, fixed in 5174d4bda48a5c4800402b54d64b7f434edfeb1c